src/app/app.component.ts

```diff
@@ -26,15 +27,28 @@ export class AppComponent {
   constructor(
     public auth: AuthService,
     private notificationService: NotificationService,
-    private snackBar: MatSnackBar
+    private snackBar: MatSnackBar,
+    private dialog: MatDialog
   ) {
     this.notificationService.notification$.subscribe(message => {
-      this.snackBar.open(message);
+      this.snackBar.open(message, 'Dismiss');
     });
     this.notificationService.error$.subscribe(errorMessage => {
-      this.snackBar.open(errorMessage);
+      this.snackBar.open(errorMessage, 'Dismiss');
+    });
+
+    this.notificationService.importantNotification$.subscribe(message => {
+      this.dialog.open(ModalComponent, {data: message});
     });
   }
   title = 'PeopleMath';
 }
 
+@Component({
```

Comment:
Any reason we're not putting this in a separate file with its own template? That's what `ng generate component` will do by default.

Reply:
I initially didn't think this would work with a component at all and then didn't stop to reevaluate once I did use a component. I created a separate file now.